src/deluge/gui/views/automation_instrument_clip_view.cpp

```diff
@@ -0,0 +1,3138 @@
+			}
+
+			ModelStackWithNoteRow* modelStackWithNoteRow =
+			    instrumentClipView.getOrCreateNoteRowForYDisplay(modelStack, instrumentClipView.lastAuditionedYDisplay);
+
+			instrumentClipView.editNoteRowLength(modelStackWithNoteRow, offset,
+			                                     instrumentClipView.lastAuditionedYDisplay);
+			//instrumentClipView.editedAnyPerNoteRowStuffSinceAuditioningBegan = true;
+			uiNeedsRendering(this);
+		}
+
+		// Unlike for all other cases where we protect against the user accidentally turning the encoder more after releasing their press on it,
+		// for this edit-NoteRow-length action, because it's a related action, it's quite likely that the user actually will want to do it after the yes-pressed-encoder-down
+		// action, which is "rotate/shift notes in row". So, we have a 250ms timeout for this one.
+		else if ((uint32_t)(AudioEngine::audioSampleTimer - instrumentClipView.timeHorizontalKnobLastReleased)
+		         >= 250 * 44) {
+			instrumentClipView.shouldIgnoreHorizontalScrollKnobActionIfNotAlsoPressedForThisNotePress = false;
+			goto wantToEditNoteRowLength;
+		}
+		return ActionResult::DEALT_WITH;
+	}
+
+	// Or, let parent deal with it
+	else {
+		ActionResult result = ClipView::horizontalEncoderAction(offset);
+		uiNeedsRendering(this);
+		setDisplayParameterNameTimer();
+		return result;
+	}
+}
+
+//new function created for automation instrument clip view to shift automations of the selected parameter
+//previously users only had the option to shift ALL automations together
+//as part of community feature i disabled automation shifting in the regular instrument clip view
+void AutomationInstrumentClipView::shiftAutomationHorizontally(int32_t offset) {
+
+	char modelStackMemory[MODEL_STACK_MAX_SIZE];
+	InstrumentClip* clip = getCurrentClip();
+	Instrument* instrument = (Instrument*)clip->output;
+
+	ModelStackWithTimelineCounter* modelStack = currentSong->setupModelStackWithCurrentClip(modelStackMemory);
+
+	ModelStackWithAutoParam* modelStackWithParam =
+	    getModelStackWithParam(modelStack, clip, clip->lastSelectedParamID, clip->lastSelectedParamType);
+
+	if (modelStackWithParam && modelStackWithParam->autoParam) {
+
+		for (int32_t xDisplay = 0; xDisplay < kDisplayWidth; xDisplay++) {
+
+			uint32_t squareStart = getPosFromSquare(xDisplay);
+
+			int32_t effectiveLength;
+
+			if (instrument->type == InstrumentType::KIT && !instrumentClipView.getAffectEntire()) {
+				ModelStackWithNoteRow* modelStackWithNoteRow = clip->getNoteRowForSelectedDrum(modelStack);
+
+				effectiveLength = modelStackWithNoteRow->getLoopLength();
+			}
+			else {
+				//this will differ for a kit when in note row mode
+				effectiveLength = clip->loopLength;
+			}
+
+			if (squareStart < effectiveLength) {
+				modelStackWithParam->autoParam->shiftHorizontally(offset, effectiveLength);
+			}
+		}
+	}
+
+	uiNeedsRendering(this);
+}
+
+//vertical encoder action
+//no change compared to instrument clip view version
```

Comment:
It would be cool if we could zoom and pan vertically in the automation view but that's probably a good thing to save for followup work